ext/phar/phar_object.c

```diff
@@ -4299,21 +4299,54 @@ static int phar_extract_file(zend_bool overwrite, phar_entry_info *entry, char *
 }
 /* }}} */
 
+static int extract_helper(phar_archive_data *archive, char *search, char *pathto, size_t pathto_len, zend_bool overwrite, char **error) { /* {{{ */
+	int extracted = 0;
+	phar_entry_info *entry;
+	int len;
+
+	/* nothing to match, or matching "/" -- extract all files */
+	len = search ? strlen(search) : 0;
+	if (0 == len || 0 == strcmp((const char *)search, "/")) {
```

Comment:
The `(const char *)` cast here shouldn't be necessary. I also want to double check if we really want to support the `"/"` case. As phar paths don't have a leading `/`, this case stands out from the others, as it's not a prefix of the path.